### ISSUE TITLE
Throttle `history.replaceState()` events?

### ISSUE THREAD

**Issue by lmenus** [storybooks/storybook#3211]
### Issue details

React DOM prevents from updating history more than 100 times in 30 seconds. Since Storybook updates history every time there is a change to the settings (e.g. knobs), it is very easy to trigger this error.

### Steps to reproduce

1/ Create a text knob.

2/ Start typing a text at least 100 characters long.

3/ Throws error if you type fast enough. I estimate the average typing speed is ~220ch based on a quick Google search, so this is not really an edge case value. A simple throttle of of 300-400ms would prevent this error.

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/react 3.3.14
- @storybook/addon-knobs 3.3.14

### Screenshots / Screencast / Code Snippets (Optional)

```js
[Error] SecurityError (DOM Exception 18): Attempt to use history.replaceState() more than 100 times per 30.000000 seconds
	rethrowCaughtError (manager.bundle.js:31337)
	processEventQueue (manager.bundle.js:31843)
	runEventQueueInBatch (manager.bundle.js:34334)
	handleTopLevel (manager.bundle.js:34343)
	handleTopLevelImpl (manager.bundle.js:34074)
	batchedUpdates (manager.bundle.js:41809)
	batchedUpdates (manager.bundle.js:33057)
	dispatchEvent (manager.bundle.js:34148)
	dispatchEvent
```

**Comment by wuweiweiwu** [comment]
Can't seem to reproduce on `addon-knobs 3.4.0-rc2` but I think we would throttle here if we needed to
https://github.com/storybooks/storybook/blob/master/lib/ui/src/modules/ui/configs/handle_routing.js#L51

**Comment by Hypnosphi** [comment]
Actually I think there's also a value in throttling the input changes themself, so that preview doesn't get the text value update too often

**Comment by dangreenisrael** [comment]
I also can't reproduce this on the latest version of master. Also, we seem to be doing some sort of throttling here already: https://github.com/storybooks/storybook/blob/a9944e1533de79ce4a1a2fd8d2b25aa450998147/addons/knobs/src/KnobManager.js#L61

**Comment by lmenus** [comment]
@wuweiweiwu @Hypnosphi @dangreenisrael I recorded a [video](https://youtu.be/aQK9Tt1iEbI) using `3.4.0-rc2`, still happens. Could this be another plugin causing it? Notice how it deletes my input (frustrating), but as soon as the errors show up I can type freely, although it won't update the text field anymore.

**Comment by dangreenisrael** [comment]
@lmenus Is that Safari?

**Comment by lmenus** [comment]
@dangreenisrael Yes, it is Safari, I cannot seem to be able to reproduce it in Chrome, although the typing behaviour is very similar (sometimes it undoes the last action)

**Comment by danielduan** [comment]
This is surprisingly not easy to do because of how the state is managed by the knobs addon.

If we throttle at the input onChange level, the inputs become uncontrolled in React and the text input becomes garbage. Same issue with changing at the form level.

Throtting at the framework `wrapStory` level will reduce the number of times the page gets rerendered, but doesn't help with the history.

We need to somehow decouple the store updates from the channel event updates. We can either add a state to each of the knob type inputs so they keep their own state and send throttled onChange updates. Or we can modify the state to dispatch two different kinds of updates, one to the input value, and the other to events channel to update the story.
